coverage/artifacts/artifacts.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package artifacts is responsible for generating and structuring artifacts
+// directory
+package artifacts
+
+import (
+	"path"
+)
+
+const (
+	//CovProfileCompletionMarker is the file name of the completion marker for coverage profiling
+	CovProfileCompletionMarker = "profile-completed"
+)
+
+//Artifacts stores information about artifacts directory and files in it
+type Artifacts struct {
+	Directory      string
+	profileName    string
+	keyProfileName string
+	covStdoutName  string
+}
+
+//New Artifact object
+func New(directory, profileName, keyProfileName, covStdoutName string) *Artifacts {
+	return &Artifacts{
```

Comment:
This _should_ fail `go vet` -- always better to use keyed struct literals:

```go

return &Artifacts{
    Directory: directory,
    profileName: profileName,
    keyProfileName: keyProfileName,
    covStdoutName: covStdoutName,
}
```

@cjwagner if this isn't failing `go vet` we should figure outwhy

Reply:
I'm not sure why this didn't fail, the package `k8s.io/test-infra/coverage/artifacts` was vetted according to the logs. `verify-govet.sh` doesn't pass against master right now though so the job is not working as intended. It seems to work locally though... I'll open an issue.

Reply:
https://github.com/kubernetes/test-infra/issues/9743

Reply:
I think we should get rid of this New function and use Artifacts{...} directly